src/sinusoid.cpp

```diff
@@ -34,60 +34,38 @@
 
 /** \author Mrinal Kalakrishnan */
 
-#include "control_toolbox/sinusoid.h"
 #include <cmath>
 #include <iostream>
 
-using namespace std;
+#include "control_toolbox/sinusoid.hpp"
 
 namespace control_toolbox
 {
-
-Sinusoid::Sinusoid(double offset, double amplitude, double frequency, double phase):
-  offset_(offset),
-  amplitude_(amplitude),
-  frequency_(frequency),
-  phase_(phase)
-{
-}
-
-bool Sinusoid::initXml(TiXmlElement *ti_xml_element)
+Sinusoid::Sinusoid(double offset, double amplitude, double frequency, double phase)
+: offset_(offset), amplitude_(amplitude), frequency_(frequency), phase_(phase)
 {
-  const char* attr;
-  attr = ti_xml_element->Attribute("offset");
-  offset_ = attr ? atof(attr) : 0.0;
-  attr = ti_xml_element->Attribute("amplitude");
-  amplitude_ = attr ? atof(attr) : 0.0;
-  attr = ti_xml_element->Attribute("frequency");
-  frequency_ = attr ? atof(attr) : 0.0;
-  attr = ti_xml_element->Attribute("phase");
-  phase_ = attr ? atof(attr) : 0.0;
-  return true; // does not fail for now, we assume a default of 0 for all params
 }
 
-Sinusoid::~Sinusoid()
-{
-}
+Sinusoid::~Sinusoid() {}
```

Comment:
empty non-virtual constructor and destructors, let's drop 'em

Reply:
Would you rather have not constructor/destructor at all? I was thinking in `default`ing them.

Reply:
not have = all defaults are defined